app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php

```diff
@@ -521,10 +521,11 @@ protected function getTemporaryTreeIndexTableName()
      *
      * Returns the name of the temporary table to use in queries.
      *
+     * @param \Magento\Store\Model\Store $store
      * @return string
      * @since 101.0.0
      */
-    protected function makeTempCategoryTreeIndex()
+    protected function makeTempCategoryTreeIndex(\Magento\Store\Model\Store $store)
```

Comment:
We can't add/remove any parameters to public and protected method, please, use guide for [backward compatible developpmenet](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)